app/assets/javascripts/services/topology_service.js

```diff
@@ -117,4 +117,23 @@ ManageIQ.angular.app.service('topologyService', function() {
     }
     return kinds
   };
+
+  // this injects some common code in the controller - temporary pending a proper merge
+  this.mixinSearch = function($scope) {
+    var topologyService = this;
+
+    $scope.searchNode = function() {
+      var svg = topologyService.getSVG($scope.d3);
+      var query = $('input#search_topology')[0].value;
```

Comment:
How about using `angular.element` here and even below?

`angular.element('input#search_topology').value`